ts/javascript/initializers.ts

```diff
@@ -0,0 +1,207 @@
+import { Initializer } from "../base.ts";
+import * as fs from "https://deno.land/std@0.201.0/fs/mod.ts";
+import { assertEquals } from "https://deno.land/std@0.201.0/assert/mod.ts";
+import outdent from "https://deno.land/x/outdent@v0.8.0/mod.ts";
+import { mapValues } from "../internal/utils.ts";
+
+const npmInitializer: Initializer = () => {
+  const existsPkgJson = fs.existsSync("package.json");
+  const existsPkgLock = fs.existsSync("package-lock.json");
+  const existsYarnLock = fs.existsSync("yarn.lock");
+  if (!existsPkgJson || (!existsPkgLock && existsYarnLock)) {
```

Comment:
Since you don't get a `package-lock.json` until after adding a dependency I felt like it's better to have our initializer just assume `npm` if you have a `package.json` and no `yarn.lock`